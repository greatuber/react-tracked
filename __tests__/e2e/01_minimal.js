/* global page */

import { toMatchImageSnapshot } from 'jest-image-snapshot';

const port = process.env.PORT || '8080';
jest.setTimeout(15 * 1000);
expect.extend({ toMatchImageSnapshot });

describe('01_minimal', () => {
  it('should work with recorded events', async () => {
    await page.goto(`http://localhost:${port}/`);

    await page.setViewport({ width: 1129, height: 324 });

    await page.waitForSelector('body > #app > div:nth-child(2) > div > button:nth-child(2)');
    await page.click('body > #app > div:nth-child(2) > div > button:nth-child(2)');
    expect(await page.screenshot()).toMatchImageSnapshot();

    await page.waitForSelector('body > #app > div:nth-child(2) > div > button:nth-child(2)');
    await page.click('body > #app > div:nth-child(2) > div > button:nth-child(2)');
    expect(await page.screenshot()).toMatchImageSnapshot();

    await page.waitForSelector('body > #app > div:nth-child(2) > div > button:nth-child(2)');
    await page.click('body > #app > div:nth-child(2) > div > button:nth-child(2)');
    expect(await page.screenshot()).toMatchImageSnapshot();

    await page.waitForSelector('body > #app > div:nth-child(2) > div > button:nth-child(3)');
    await page.click('body > #app > div:nth-child(2) > div > button:nth-child(3)');
    expect(await page.screenshot()).toMatchImageSnapshot();

    await page.waitForSelector('body > #app > div:nth-child(2) > div > button:nth-child(3)');
    await page.click('body > #app > div:nth-child(2) > div > button:nth-child(3)');
    expect(await page.screenshot()).toMatchImageSnapshot();

    await page.waitForSelector('body > #app > div:nth-child(2) > div > button:nth-child(3)');
    await page.click('body > #app > div:nth-child(2) > div > button:nth-child(3)');
    expect(await page.screenshot()).toMatchImageSnapshot();

    await page.waitForSelector('body > #app > div:nth-child(3) > div > button:nth-child(3)');
    await page.click('body > #app > div:nth-child(3) > div > button:nth-child(3)');
    expect(await page.screenshot()).toMatchImageSnapshot();

    await page.waitForSelector('body > #app > div:nth-child(3) > div > button:nth-child(3)');
    await page.click('body > #app > div:nth-child(3) > div > button:nth-child(3)');
    expect(await page.screenshot()).toMatchImageSnapshot();

    await page.waitForSelector('body > #app > div:nth-child(3) > div > button:nth-child(3)');
    await page.click('body > #app > div:nth-child(3) > div > button:nth-child(3)');
    expect(await page.screenshot()).toMatchImageSnapshot();

    await page.waitForSelector('body > #app > div:nth-child(3) > div > button:nth-child(2)');
    await page.click('body > #app > div:nth-child(3) > div > button:nth-child(2)');
    expect(await page.screenshot()).toMatchImageSnapshot();

    await page.waitForSelector('body > #app > div:nth-child(3) > div > button:nth-child(2)');
    await page.click('body > #app > div:nth-child(3) > div > button:nth-child(2)');
    expect(await page.screenshot()).toMatchImageSnapshot();

    await page.waitForSelector('body > #app > div:nth-child(3) > div > button:nth-child(2)');
    await page.click('body > #app > div:nth-child(3) > div > button:nth-child(2)');
    await page.waitForSelector('body > #app > div:nth-child(5) > div > input');
    await page.click('body > #app > div:nth-child(5) > div > input');
    expect(await page.screenshot()).toMatchImageSnapshot();

    await page.type('body > #app > div:nth-child(5) > div > input', '1');
    expect(await page.screenshot()).toMatchImageSnapshot();

    await page.type('body > #app > div:nth-child(5) > div > input', '2');
    expect(await page.screenshot()).toMatchImageSnapshot();

    await page.type('body > #app > div:nth-child(5) > div > input', '3');
    expect(await page.screenshot()).toMatchImageSnapshot();
  });
});
