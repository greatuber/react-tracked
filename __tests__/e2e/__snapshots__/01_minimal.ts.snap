// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`01_minimal should work with recorded events 1`] = `
"
    <div id=\\"app\\"><h1>Counter</h1><div>numRendered: 18<div><span>Count: 1</span><button type=\\"button\\">+1</button><button type=\\"button\\">-1</button></div></div><div>numRendered: 20<div><span>Count: 1</span><button type=\\"button\\">+1</button><button type=\\"button\\">-1</button></div></div><h1>TextBox</h1><div>numRendered: 6<div><span>Text: hello</span><input value=\\"hello\\"></div></div><div>numRendered: 8<div><span>Text: hello</span><input value=\\"hello\\"></div></div></div>
  

"
`;

exports[`01_minimal should work with recorded events 2`] = `
"
    <div id=\\"app\\"><h1>Counter</h1><div>numRendered: 30<div><span>Count: 2</span><button type=\\"button\\">+1</button><button type=\\"button\\">-1</button></div></div><div>numRendered: 32<div><span>Count: 2</span><button type=\\"button\\">+1</button><button type=\\"button\\">-1</button></div></div><h1>TextBox</h1><div>numRendered: 6<div><span>Text: hello</span><input value=\\"hello\\"></div></div><div>numRendered: 8<div><span>Text: hello</span><input value=\\"hello\\"></div></div></div>
  

"
`;

exports[`01_minimal should work with recorded events 3`] = `
"
    <div id=\\"app\\"><h1>Counter</h1><div>numRendered: 42<div><span>Count: 3</span><button type=\\"button\\">+1</button><button type=\\"button\\">-1</button></div></div><div>numRendered: 44<div><span>Count: 3</span><button type=\\"button\\">+1</button><button type=\\"button\\">-1</button></div></div><h1>TextBox</h1><div>numRendered: 6<div><span>Text: hello</span><input value=\\"hello\\"></div></div><div>numRendered: 8<div><span>Text: hello</span><input value=\\"hello\\"></div></div></div>
  

"
`;

exports[`01_minimal should work with recorded events 4`] = `
"
    <div id=\\"app\\"><h1>Counter</h1><div>numRendered: 54<div><span>Count: 2</span><button type=\\"button\\">+1</button><button type=\\"button\\">-1</button></div></div><div>numRendered: 56<div><span>Count: 2</span><button type=\\"button\\">+1</button><button type=\\"button\\">-1</button></div></div><h1>TextBox</h1><div>numRendered: 6<div><span>Text: hello</span><input value=\\"hello\\"></div></div><div>numRendered: 8<div><span>Text: hello</span><input value=\\"hello\\"></div></div></div>
  

"
`;

exports[`01_minimal should work with recorded events 5`] = `
"
    <div id=\\"app\\"><h1>Counter</h1><div>numRendered: 66<div><span>Count: 1</span><button type=\\"button\\">+1</button><button type=\\"button\\">-1</button></div></div><div>numRendered: 68<div><span>Count: 1</span><button type=\\"button\\">+1</button><button type=\\"button\\">-1</button></div></div><h1>TextBox</h1><div>numRendered: 6<div><span>Text: hello</span><input value=\\"hello\\"></div></div><div>numRendered: 8<div><span>Text: hello</span><input value=\\"hello\\"></div></div></div>
  

"
`;

exports[`01_minimal should work with recorded events 6`] = `
"
    <div id=\\"app\\"><h1>Counter</h1><div>numRendered: 78<div><span>Count: 0</span><button type=\\"button\\">+1</button><button type=\\"button\\">-1</button></div></div><div>numRendered: 80<div><span>Count: 0</span><button type=\\"button\\">+1</button><button type=\\"button\\">-1</button></div></div><h1>TextBox</h1><div>numRendered: 6<div><span>Text: hello</span><input value=\\"hello\\"></div></div><div>numRendered: 8<div><span>Text: hello</span><input value=\\"hello\\"></div></div></div>
  

"
`;

exports[`01_minimal should work with recorded events 7`] = `
"
    <div id=\\"app\\"><h1>Counter</h1><div>numRendered: 90<div><span>Count: -1</span><button type=\\"button\\">+1</button><button type=\\"button\\">-1</button></div></div><div>numRendered: 92<div><span>Count: -1</span><button type=\\"button\\">+1</button><button type=\\"button\\">-1</button></div></div><h1>TextBox</h1><div>numRendered: 6<div><span>Text: hello</span><input value=\\"hello\\"></div></div><div>numRendered: 8<div><span>Text: hello</span><input value=\\"hello\\"></div></div></div>
  

"
`;

exports[`01_minimal should work with recorded events 8`] = `
"
    <div id=\\"app\\"><h1>Counter</h1><div>numRendered: 102<div><span>Count: -2</span><button type=\\"button\\">+1</button><button type=\\"button\\">-1</button></div></div><div>numRendered: 104<div><span>Count: -2</span><button type=\\"button\\">+1</button><button type=\\"button\\">-1</button></div></div><h1>TextBox</h1><div>numRendered: 6<div><span>Text: hello</span><input value=\\"hello\\"></div></div><div>numRendered: 8<div><span>Text: hello</span><input value=\\"hello\\"></div></div></div>
  

"
`;

exports[`01_minimal should work with recorded events 9`] = `
"
    <div id=\\"app\\"><h1>Counter</h1><div>numRendered: 114<div><span>Count: -3</span><button type=\\"button\\">+1</button><button type=\\"button\\">-1</button></div></div><div>numRendered: 116<div><span>Count: -3</span><button type=\\"button\\">+1</button><button type=\\"button\\">-1</button></div></div><h1>TextBox</h1><div>numRendered: 6<div><span>Text: hello</span><input value=\\"hello\\"></div></div><div>numRendered: 8<div><span>Text: hello</span><input value=\\"hello\\"></div></div></div>
  

"
`;

exports[`01_minimal should work with recorded events 10`] = `
"
    <div id=\\"app\\"><h1>Counter</h1><div>numRendered: 126<div><span>Count: -2</span><button type=\\"button\\">+1</button><button type=\\"button\\">-1</button></div></div><div>numRendered: 128<div><span>Count: -2</span><button type=\\"button\\">+1</button><button type=\\"button\\">-1</button></div></div><h1>TextBox</h1><div>numRendered: 6<div><span>Text: hello</span><input value=\\"hello\\"></div></div><div>numRendered: 8<div><span>Text: hello</span><input value=\\"hello\\"></div></div></div>
  

"
`;

exports[`01_minimal should work with recorded events 11`] = `
"
    <div id=\\"app\\"><h1>Counter</h1><div>numRendered: 138<div><span>Count: -1</span><button type=\\"button\\">+1</button><button type=\\"button\\">-1</button></div></div><div>numRendered: 140<div><span>Count: -1</span><button type=\\"button\\">+1</button><button type=\\"button\\">-1</button></div></div><h1>TextBox</h1><div>numRendered: 6<div><span>Text: hello</span><input value=\\"hello\\"></div></div><div>numRendered: 8<div><span>Text: hello</span><input value=\\"hello\\"></div></div></div>
  

"
`;

exports[`01_minimal should work with recorded events 12`] = `
"
    <div id=\\"app\\"><h1>Counter</h1><div>numRendered: 150<div><span>Count: 0</span><button type=\\"button\\">+1</button><button type=\\"button\\">-1</button></div></div><div>numRendered: 152<div><span>Count: 0</span><button type=\\"button\\">+1</button><button type=\\"button\\">-1</button></div></div><h1>TextBox</h1><div>numRendered: 6<div><span>Text: hello</span><input value=\\"hello\\"></div></div><div>numRendered: 8<div><span>Text: hello</span><input value=\\"hello\\"></div></div></div>
  

"
`;

exports[`01_minimal should work with recorded events 13`] = `
"
    <div id=\\"app\\"><h1>Counter</h1><div>numRendered: 150<div><span>Count: 0</span><button type=\\"button\\">+1</button><button type=\\"button\\">-1</button></div></div><div>numRendered: 152<div><span>Count: 0</span><button type=\\"button\\">+1</button><button type=\\"button\\">-1</button></div></div><h1>TextBox</h1><div>numRendered: 6<div><span>Text: hello</span><input value=\\"hello\\"></div></div><div>numRendered: 8<div><span>Text: hello</span><input value=\\"hello\\"></div></div></div>
  

"
`;

exports[`01_minimal should work with recorded events 14`] = `
"
    <div id=\\"app\\"><h1>Counter</h1><div>numRendered: 150<div><span>Count: 0</span><button type=\\"button\\">+1</button><button type=\\"button\\">-1</button></div></div><div>numRendered: 152<div><span>Count: 0</span><button type=\\"button\\">+1</button><button type=\\"button\\">-1</button></div></div><h1>TextBox</h1><div>numRendered: 162<div><span>Text: hello1</span><input value=\\"hello1\\"></div></div><div>numRendered: 164<div><span>Text: hello1</span><input value=\\"hello1\\"></div></div></div>
  

"
`;

exports[`01_minimal should work with recorded events 15`] = `
"
    <div id=\\"app\\"><h1>Counter</h1><div>numRendered: 150<div><span>Count: 0</span><button type=\\"button\\">+1</button><button type=\\"button\\">-1</button></div></div><div>numRendered: 152<div><span>Count: 0</span><button type=\\"button\\">+1</button><button type=\\"button\\">-1</button></div></div><h1>TextBox</h1><div>numRendered: 174<div><span>Text: hello12</span><input value=\\"hello12\\"></div></div><div>numRendered: 176<div><span>Text: hello12</span><input value=\\"hello12\\"></div></div></div>
  

"
`;

exports[`01_minimal should work with recorded events 16`] = `
"
    <div id=\\"app\\"><h1>Counter</h1><div>numRendered: 150<div><span>Count: 0</span><button type=\\"button\\">+1</button><button type=\\"button\\">-1</button></div></div><div>numRendered: 152<div><span>Count: 0</span><button type=\\"button\\">+1</button><button type=\\"button\\">-1</button></div></div><h1>TextBox</h1><div>numRendered: 186<div><span>Text: hello123</span><input value=\\"hello123\\"></div></div><div>numRendered: 188<div><span>Text: hello123</span><input value=\\"hello123\\"></div></div></div>
  

"
`;
